### DISCUSSION TITLE
build Vertex Label index

### DISCUSSION THREAD

**Discussion by galaxyknight** [JanusGraph/janusgraph#3250] [Q&A]
How to build index for Vertex Label？

**Comment by vtslab** [comment]
Usually, an index is meant to return just a small subset of the dataset. An index on a vertex label would typically return a large subset. Probably, you are better off with a label constraint on a property index, see: https://docs.janusgraph.org/schema/index-management/index-performance/#label-constraint

**Reply by galaxyknight** [reply]
Thank you for your reply. In such a scenario, for example, I have a large-scale book KG, and I want to query how many computer books there are. My gremlin query is: "g.V(). hasLabel ('ComputerScience '). count()". In this case, I have configured a label constraint for the attribute, which does not seem to trigger index acceleration. How can I speed up my search in this situation

**Reply by vtslab** [reply]
In this case you would use 'Book' as a label and 'ComputerScience' as a value of a 'topic' property. In the index for the 'topic' propertykey you could add a label constraint (if other vertex types with the same propertykey do not need to be indexed).
